src/TritonCTS/src/DbWrapper.cpp

```diff
@@ -74,25 +77,60 @@ void DbWrapper::initDB() {
 
 void DbWrapper::initAllClocks() {
         std::cout << " Initializing clock nets\n";
-        
+
+        clearNumClocks();
+
+        std::cout << " Looking for clock nets in the design\n";
+
+        //Uses dbSta to find all clock nets in the design.
+
+        std::set<odb::dbNet*> clockNets;
+
+        _openSta->findClkNets(clockNets);
+
+        //Checks the user input in case there are other nets that need to be added to the set.
         std::vector<std::string> clockNetNames;
         parseClockNames(clockNetNames);
 
-        std::cout << " Looking for clock nets in the design\n";
         for (const std::string& name: clockNetNames) {
                 odb::dbNet* net = _block->findNet(name.c_str());
                 if (!net) {
                         odb::dbITerm* iterm = _block->findITerm(name.c_str());
+                        if (!iterm) {
+                                std::cout << " [WARNING] Net \"" << name << "\" not in design. Skipping...\n";
+                                continue;
+                        }
                         net = iterm->getNet();
                 }
                 if (!net) {
                         std::cout << " [WARNING] Net \"" << name << "\" not in design. Skipping...\n";
                         continue;
+                }
+                //Since a set is unique, only the nets not found by dbSta are added.
+                clockNets.insert(net);
+        }
+        for (odb::dbNet* net : clockNets){
+                if (!net) {
+                        odb::dbITerm* iterm = _block->findITerm(net->getName().c_str());
```

Comment:
This is silly. _openSta->findClkNets returns dbNets and this loop find an dbITerm using the net's **name** and then gets the net back from the iterm. It should not be doing either. It is pointless and ill advised for the nets returned by findClkNets and redundant from the clock name loop (why does that even exist?) also.

Reply:
This was changed in the last commit.